server/src/app/GaWorker.ts

```diff
@@ -76,12 +76,12 @@ export class GaWorker {
     for (var index = 0; index < bot.melody.notes.length; index++) {
       if (Math.random() < GaWorker.MUTATION_RATE) {
         const notes = this.getPossibleNotesFromRules(index, bot);
-
+        const weights = this.assignNoteWeights(index, notes, bot.melody);
         if (!notes.length) {
           return;
         }
 
-        bot.melody.notes[index] = selectRandom(notes);
```

Comment:
Let's use `selectRandom` right now to change less variables while we get testing. Comment out the `selectRandomWeighted` though so we can feed it in later :) .